src/Metadata/PropertyTypeDateTime.php

```diff
@@ -67,18 +67,37 @@ public function getDeserializeFormat(): ?string
         return null;
     }
 
+    public function getDeserializeFormats(): ?array
+    {
+        if ($this->dateTimeOptions) {
+            return $this->dateTimeOptions->getDeserializeFormats();
+        }
+
+        return null;
+    }
+
     public function merge(PropertyType $other): PropertyType
     {
         $nullable = $this->isNullable() && $other->isNullable();
 
         if ($other instanceof PropertyTypeUnknown) {
             return new self($this->immutable, $nullable, $this->dateTimeOptions);
         }
+        if (($other instanceof PropertyTypeClass) && is_a($other->getClassName(), \DateTimeInterface::class, true)) {
+            if (is_a($other->getClassName(), \DateTimeImmutable::class, true)) {
+                return new self(true, $nullable, $this->dateTimeOptions);
```

Comment:
can't a DateTimeInterface property type also have options configured?

Reply:
how would we process an annotation `\DateTimeInterface<Y-m-d>`? i think we should first check if the other thing is instanceof self and do the merge, then early return.
and only otherwise check if it is a property of DateTimeInterface (as created by e.g. the php parser)

Reply:
ah, the PropertyTypeDateTime does not extend PropertyTypeClass, so the order does not matter and its fine here.